### PR TITLE
deps: Bump `com_github_nghttp2_nghttp2` -> 1.52.0

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
cc @yanavlasov

**Comment by phlax** [comment]
not sure why this is failing - passes locally - seems like a transient flake but i dont seem to trigger it any other way - v odd

**Comment by phlax** [comment]
> v odd

works today 8)

File: ci/do_ci.sh [hunk 1]
```diff
@@ -509,6 +509,7 @@ elif [[ "$CI_TARGET" == "deps" ]]; then
   bazel run "${BAZEL_BUILD_OPTIONS[@]}" //tools/dependency:check \
         --action_env=TODAY_DATE \
         -- -v warn \
+           -l debug \
```
Comment (htuch) [inline ci/do_ci.sh:512]: Did you want to keep this?

**Review by htuch — APPROVED** [review]

File: ci/do_ci.sh (hunk 1 above)
Comment (phlax) [reply ci/do_ci.sh:512]: good spot, no

Comment (phlax) [reply ci/do_ci.sh]: removed

**Review by htuch — APPROVED** [review]

**Pull request merged**